build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -0,0 +1,453 @@
+            const minVer = semver.minVersion(depVersion);
+            if (minVer === null) {
+                throw new Error(`semver.minVersion was null: ${depVersion} (${depName})`);
+            }
+
+            // If the min version has a pre-release section, then it needs to be released.
+            if (minVer.prerelease.length > 0) {
+                const depPkg = context.fullPackageMap.get(depName);
+                if (depPkg === undefined) {
+                    throw new Error(`Can't find package in context: ${depName}`);
+                }
+
+                const nameToUse =
+                    depPkg.monoRepo === undefined ? depPkg.name : depPkg.monoRepo.kind;
+                prereleasePackages.set(nameToUse, depVersion);
+
+                if (depPkg.monoRepo === undefined) {
+                    prereleasePackages.set(depPkg.name, depVersion);
+                } else {
+                    prereleaseGroups.set(depPkg.monoRepo.kind, depVersion);
+                }
+            }
+        }
+    }
+
+    const isEmpty = prereleaseGroups.size === 0 && prereleasePackages.size === 0;
+    return {
+        releaseGroups: prereleaseGroups,
+        packages: prereleasePackages,
+        isEmpty,
+    };
+}
+
+/** Convenience function to hydrate an array of {@link Package} objects from {@link ReleasePackage}s. */
+function getPackagesFromReleasePackages(
+    context: Context,
+    relPackages: ReleasePackage[],
+): Package[] {
+    const packages: Package[] = [];
+
+    for (const rp of relPackages) {
+        const pkg = context.fullPackageMap.get(rp);
+        if (pkg === undefined) {
+            throw new Error(`Can't find package in context: ${rp}`);
+        }
+
+        packages.push(pkg);
+    }
+
+    return packages;
+}
+
+/**
+ * Returns true if a release group or package in the repo has been released.
+ *
+ * @param context - The context.
+ * @param releaseGroupOrPackage - The release group to check.
+ * @returns True if the release group was released.
+ *
+ * @remarks
+ *
+ * This function exclusively uses the tags in the repo to determine whether a release has bee done or not.
+ *
+ * @internal
+ */
+export async function isReleased(
+    context: Context,
+    releaseGroupOrPackage: MonoRepo | Package | string,
+    version: string,
+    log?: Logger,
+): Promise<boolean> {
+    await context.gitRepo.fetchTags();
+
+    const tagName = generateReleaseTagName(releaseGroupOrPackage, version);
+    if (typeof releaseGroupOrPackage === "string" && isReleaseGroup(releaseGroupOrPackage)) {
+        // eslint-disable-next-line no-param-reassign
+        releaseGroupOrPackage = context.repo.releaseGroups.get(releaseGroupOrPackage)!;
+    }
+
+    log?.verbose(`Checking for tag '${tagName}'`);
+    const rawTag = await context.gitRepo.getTags(tagName);
+    return rawTag.trim() === tagName;
+}
+
+/**
+ * Generates the correct tag name for the release of a given release group and version.
```

Comment:
It might be worth explicitly calling out that this is a git tag. On first read it wasn't clear to me.,